src/mpid/ch3/src/mpid_startall.c

```diff
@@ -55,16 +55,16 @@ int MPID_Startall(int count, MPIR_Request * requests[])
     {
 	MPIR_Request * const preq = requests[i];
 
-        /* continue if the source/dest is MPI_PROC_NULL */
-        if (preq->dev.match.parts.rank == MPI_PROC_NULL)
-            continue;
-
         if (preq->kind == MPIR_REQUEST_KIND__PREQUEST_COLL) {
             mpi_errno = MPIR_Persist_coll_start(preq);
             MPIR_ERR_CHECK(mpi_errno);
             continue;
         }
 
+        /* continue if the source/dest is MPI_PROC_NULL */
```

Comment:
Add an assertion to make the assumption clear?

Reply:
Added.